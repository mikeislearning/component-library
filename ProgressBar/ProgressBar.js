import React from 'react';
import { Text, View } from 'react-native';
import { PropTypes } from 'prop-types';

import styles from './ProgressBar.styles';

export default class ProgressBar extends React.Component {
  static defaultProps = {
    currentStep: 0,
  };

  static propTypes = {
    currentStep: PropTypes.number,
  };

  render() {
    return (
      <View style={styles(this.props.currentStep).container}>
        <View style={styles(this.props.currentStep).progressBar}>
          <View style={styles(this.props.currentStep).currentProgressBar} />
        </View>
        <Text style={styles(this.props.currentStep).progressBarText}>
          {this.props.currentStep}/7
        </Text>
      </View>
    );
  }
}
