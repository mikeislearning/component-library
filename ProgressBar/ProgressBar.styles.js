import { StyleSheet } from 'react-native';
import { colors } from 'app/theme';

const styles = (currentProgress) =>
  StyleSheet.create({
    container: {
      alignItems: 'center',
      marginBottom: 30,
    },
    progressBar: {
      width: 280,
      height: 10,
      backgroundColor: colors.gray.light,
      borderRadius: 25,
    },
    currentProgressBar: {
      width: currentProgress * 40,
      height: 10,
      backgroundColor: colors.red.medium,
      borderBottomLeftRadius: 25,
      borderTopLeftRadius: 25,
    },
    progressBarText: {
      fontSize: 16,
    },
  });

export default styles;
