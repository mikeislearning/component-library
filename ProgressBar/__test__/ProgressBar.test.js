/**
 * @format
 */

import React from 'react';
import { render } from 'react-native-testing-library';

import ProgressBar from '../ProgressBar';

describe('<ProgressBar />', () => {
  it('renders correctly', () => {
    const progressBar = render(<ProgressBar currentStep={1} />);
    expect(progressBar).toMatchSnapshot();
  });
});
