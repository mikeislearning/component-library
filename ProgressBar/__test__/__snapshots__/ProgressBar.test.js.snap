// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 30,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ECECEC",
        "borderRadius": 25,
        "height": 10,
        "width": 280,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#CC2E41",
          "borderBottomLeftRadius": 25,
          "borderTopLeftRadius": 25,
          "height": 10,
          "width": 40,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    1
    /7
  </Text>
</View>
`;
